comit/src/network/protocols/orderbook.rs

```diff
@@ -229,37 +229,36 @@ pub enum OrderbookError {
     #[error("could not unsubscribe to all peers for topic")]
     UnSubscribe,
 }
+#[derive(Clone, Copy, Debug, Serialize, Deserialize, PartialEq, Eq, Hash)]
+pub enum TradeDirection {
+    Buy,
+    Sell,
+}
 
 #[derive(Clone, Copy, Debug, Serialize, Deserialize, PartialEq, Eq, Hash)]
-pub struct TradingPair {
-    pub buy: SwapType,
-    pub sell: SwapType,
+pub enum TradingPair {
+    BtcDai,
 }
 
 impl TradingPair {
-    pub fn to_topic(&self, peer: &PeerId) -> Topic {
-        let trading_pair_topic = TradingPairTopic {
-            peer: PeerId::into_bytes(peer.clone()),
-            buy: self.buy,
-            sell: self.sell,
-        };
-        trading_pair_topic.to_topic()
+    pub fn to_topic(&self) -> Topic {
+        match self {
+            TradingPair::BtcDai => Topic::new("BTC_DAI".to_string()),
+        }
     }
 }
 
 #[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq, Hash)]
-pub struct TradingPairTopic {
-    peer: Vec<u8>,
-    buy: SwapType,
-    sell: SwapType,
+pub struct TradingPairFromMaker {
```

Comment:
Can I suggest that you just delete all this code that maps peer ids of a maker to trading pairs they are offering. Correct me if I have misunderstood the code but
- We only support two topics 1. "makers" 2. "BTC_DAI"
- We only support two nodes, there is no 'network'
- Any logic that needs to be done before taking an order (e.g. whitelisting/blacklisting of peers) can be done on a order level, it is not done on a topic level.

What use is mapping peer id to topic?

So unless I'm missing something, and please correct me if I am, this mapping code actually achieves nothing.

Reply:
Pardon me of chiming in here but:

> We only support two nodes, there is no 'network'

The orderbook supports more than two nodes. The orderbook is in-fact a network of nodes gossiping around orders.

> Any logic that needs to be done before taking an order (e.g. whitelisting/blacklisting of peers) can be done on a order level, it is not done on a topic level.

If we don't go for explicit subscriptions to makers, then the whole "makers" topic is also redundant because you can just subscribe to "BTC_DAI" directly.

The only thing I'd ask for if you want to drop this functionality is that you actually make it an implementation detail, i.e. no part of this design must leak out of the orderbook NB and a transition to separate topics should not require an API change :)